Assignment 3.Rmd

```diff
@@ -127,29 +134,178 @@ Now we can plot our line plot using the ggplot command, "ggplot()".
+
+K10$cluster <- as.factor(K10$cluster)
+
+ggplot(K10, aes(week, avg_motiv, color = cluster)) + geom_line()+xlab("Week")+ylab("Avg, Motivation")
+
+```
 
 ##Part II
 
 Using the data collected for Assignment 2 (which classes students were in), cluster the students, then redraw the graph of the class but color the students according the cluster they are in.
 
+```{r}
+
+H1 <- read.csv("hudk4050-classes.csv", header = TRUE)
+H1 <- filter(H1, First.Name != "ZIMO")
+
+H1 <- tidyr::unite(H1, Name, `Last.Name`, `First.Name`, sep = " ", remove = TRUE)
+
+H2 <- tidyr::gather(H1, course.label, course, `Class.1`, `Class.2`, `Class.3`, `Class.4`, `Class.5`, Class.6, na.rm = TRUE, convert = FALSE)
+
+H2 <- dplyr::select(H2, Name, course)
+H2$course = gsub(" ", "", H2$course)
+H2 <- dplyr::filter(H2, course > 0, course != "HUDK4050")
+
+H2$Count <- 1
+
+H3 <- tidyr::spread(H2, course, Count)
+
+row.names(H3) <- H3$Name
+H3$Name <- NULL
+H3 <- ifelse(is.na(H3), 0, 1)
+
+H4 <- as.matrix(H3)
+
+H5 <- H4 %*% t(H4)
+diag(H5) <- NA
+library(klaR)
+
+fit <- kmodes(H3, 5, iter.max = 10, weighted = FALSE)
+fit$cluster
+H3a = data.frame(H3, fit$cluster)
+
+library(igraph)
+H <- graph.adjacency(H5,mode="undirected")
+
+plot(H,layout=layout.fruchterman.reingold, vertex.size=10, vertex.color=H3a$fit.cluster)
+
+```
+
 ##Part III
 
 In class activity 6 you clustered students in the class by the answers to a questionaire. Create a visualization that shows the overlap between these clusters and the clusters generated in part II.
 
+##Mosaic plot
+
+```{r}
+library(tidyr)
+library(dplyr)
+```
+
+```{r}
+DF1 <- read.csv("HUDK405019-clustering.csv", header = TRUE)
+
+DF1 <- unite(DF1, "Name", c("First.Name", "Last.Name"), sep = ".")
+
+row.names(DF1) <- DF1$Name
+
+DF1$Name <- NULL
+
+DF2 <- select(DF1, 1:11)
+
+DF2 <- DF2 %>% mutate_all(funs(gsub("[a-zA-Z]", "", .)))
+
+DF2 <- DF2 %>% mutate_all(funs(as.numeric(.)))
+
+DF2 <- as.data.frame(scale(DF2))
+
+DF2 <- DF2 %>% mutate_all(funs(ifelse(is.na(.) == TRUE, 0, .)))
+
+```
+
+```{r}
+
+install.packages("ggmap")
+install.packages("rgdal")
+library(ggmap)
+library(tmaptools)
+```
+
+```{r}
+
+DF3 <- select(DF1, 13:14)
+
+names(DF3) <- c("lattitude", "longitude")
+
+DF3 <- DF3 %>% mutate_all(funs(gsub("[a-zA-Z]", "", .)))
+DF3 <- DF3 %>% mutate_all(funs(sub("[?]", "", .)))
+
+DF3$lattitude <- sub(",.*$","", DF3$lattitude) 
+DF3$lattitude <- sub("°.*$","", DF3$lattitude)
+
+DF3$longitude <- gsub(".*,","",DF3$longitude)
+DF3$longitude <- sub("°.*$","", DF3$longitude)
+
+DF3 <- DF3 %>% mutate_all(funs(as.numeric(.)))
+
+
+```
+
+```{r}
+
+fit <- kmeans(DF2, 3) 
+
+fit$cluster
+
+DF4 <- data.frame(DF2, DF3, fit$cluster)
+
+
+```
+
+```{r}
+
+ggplot(DF4, aes(longitude, lattitude, color = as.factor(fit.cluster))) + geom_point(size = 3)
+
+```
+
+```{r}
+Cluster_Region= DF4$fit.cluster
+Cluster_Classes = H3a$fit.cluster 
+library(vcd)
+P1 <- structable(Cluster_Classes~Cluster_Region)
+mosaic(P1,shade=TRUE, legend=TRUE)
+```
+
```

Comment:
Good use of the mosaic plot! Overall, great job on the coding aspect of this assignment but you may want to work on the analysis part of this assignment. Nonetheless, keep up the good work!